Lib/test/test_capi/test_float.py

```diff
@@ -183,8 +183,14 @@ def test_pack_unpack_roundtrip(self):
     def test_pack_unpack_roundtrip_for_nans(self):
         pack = _testcapi.float_pack
         unpack = _testcapi.float_unpack
+        arch_32bit = (sys.maxsize == 2147483647)
+        if (not arch_32bit) or (sys.platform == 'win32'):
+            sizes = (2, 4, 8)
+        else:
+            sizes = (8,)
+
         for _ in range(1000):
-            for size in (2, 4, 8):
+            for size in sizes:
```

Comment:
Why this?

Do we have test failures, which not related to nan's type bit?

Reply:
Tests still fail with 16-bit and 32-bit floats.